File: 0622. Design Circular Queue/Solution.java
class MyCircularQueue {
    int[] queue;
    int front = 0, rear = 1, len;

    public MyCircularQueue(int k) {
        queue = new int[k + 1];
        len = k + 1;
    }

    public boolean enQueue(int value) {
        if (isFull()) {
            return false;
        } else {
            queue[rear] = value;
            rear = (rear + 1) % len;
            return true;
        }
    }

    public boolean deQueue() {
        if (isEmpty()) {
            return false;
        } else {
            front = (front + 1) % len;
            return true;
        }
    }

    public int Front() {
        if (isEmpty()) {
            return -1;
        } else {
            return queue[(front + 1) % len];
        }
    }

    public int Rear() {
        if (isEmpty()) {
            return -1;
        } else {
            return queue[(rear - 1 + len) % len];
        }
    }

    public boolean isEmpty() {
        return (front + 1) % len == rear;
    }

    public boolean isFull() {
        return front == rear;
    }
}

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue obj = new MyCircularQueue(k);
 * boolean param_1 = obj.enQueue(value);
 * boolean param_2 = obj.deQueue();
 * int param_3 = obj.Front();
 * int param_4 = obj.Rear();
 * boolean param_5 = obj.isEmpty();
 * boolean param_6 = obj.isFull();
 */

File: 0622. Design Circular Queue/Solution.py
class MyCircularQueue:

    def __init__(self, k: int):
        self.front = 0
        self.rear = 1
        self.len = k + 1
        self.queue = [0] * self.len

    def enQueue(self, value: int) -> bool:
        if self.isFull():
            return False
        self.queue[self.rear] = value
        self.rear = (self.rear + 1) % self.len
        return True

    def deQueue(self) -> bool:
        if self.isEmpty():
            return False
        self.front = (self.front + 1) % self.len
        return True

    def Front(self) -> int:
        if self.isEmpty():
            return -1
        return self.queue[(self.front + 1) % self.len]

    def Rear(self) -> int:
        if self.isEmpty():
            return -1
        return self.queue[(self.rear - 1 + self.len) % self.len]

    def isEmpty(self) -> bool:
        return (self.front + 1) % self.len == self.rear

    def isFull(self) -> bool:
        return self.front == self.rear


# Your MyCircularQueue object will be instantiated and called as such:
# obj = MyCircularQueue(k)
# param_1 = obj.enQueue(value)
# param_2 = obj.deQueue()
# param_3 = obj.Front()
# param_4 = obj.Rear()
# param_5 = obj.isEmpty()
# param_6 = obj.isFull()
